Source/WebCore/PAL/pal/text/TextEncoding.cpp

```diff
@@ -176,7 +184,7 @@ const TextEncoding& UTF16LittleEndianEncoding()
 
 const TextEncoding& UTF8Encoding()
 {
-    static NeverDestroyed<TextEncoding> globalUTF8Encoding("UTF-8"_s);
+    static NeverDestroyed<TextEncoding> globalUTF8Encoding(utf8Atom());
```

Comment:
Why is this better? It was calling the `TextEncoding` constructing that was taking an ASCIILiteral. It shouldn't have constructed a `String`.

Reply:
It was definitely constructing a `StringImpl`. But perhaps it was the first time AtomString had seem `UTF-8` in the process since launch?

Reply:
Can I see the backtrace of the StringImpl construction? I'm curious where it's at. It's not obvious to me. `"UTF-8"_s` constructs an ASCIILiteral and TextEncoder stores an ASCIILiteral. There should be no need to construct a StringImpl at all?